tests/generic_views/test_edit.py

```diff
@@ -426,3 +426,84 @@ def test_delete_without_redirect(self):
         # get_absolute_url provided
         with self.assertRaises(ImproperlyConfigured):
             self.client.post('/edit/author/%d/delete/naive/' % a.pk)
+
+    def test_delete_with_form_as_post(self):
+        a = Author.objects.create(
+            name='Randall Munroe',
+            slug='randall-munroe',
+        )
+
+        res = self.client.get('/edit/author/%d/delete/form/' % a.pk)
+        self.assertEqual(res.status_code, 200)
+        self.assertEqual(res.context['object'], Author.objects.get(pk=a.pk))
+        self.assertEqual(res.context['author'], Author.objects.get(pk=a.pk))
+        self.assertTemplateUsed(res, 'generic_views/author_confirm_delete.html')
+
+        res = self.client.post('/edit/author/%d/delete/form/' % a.pk,
+                               data={"confirm": True})
+        self.assertEqual(res.status_code, 302)
+        self.assertRedirects(res, 'http://testserver/list/authors/')
+        self.assertQuerysetEqual(Author.objects.all(), [])
+
+    def test_delete_with_form_as_post_with_validation_error(self):
+        a = Author.objects.create(
+            name='Randall Munroe',
+            slug='randall-munroe',
+        )
+
+        res = self.client.get('/edit/author/%d/delete/form/' % a.pk)
+        self.assertEqual(res.status_code, 200)
+        self.assertEqual(res.context['object'], Author.objects.get(pk=a.pk))
+        self.assertEqual(res.context['author'], Author.objects.get(pk=a.pk))
+        self.assertTemplateUsed(res, 'generic_views/author_confirm_delete.html')
+
+        res = self.client.post('/edit/author/%d/delete/form/' % a.pk)
+        self.assertEqual(res.status_code, 200)
+        self.assertEqual(len(res.context_data["form"].errors), 2)
+        self.assertEqual(res.context_data["form"].errors["__all__"],
```

Comment:
We prefer hanging indent style like this:
```
self.assertEqual(
    res.context_data["form"].errors["__all__"],
    ['You must confirm the delete.'])
)
```
Also please drop the u prefix on strings.